### ISSUE TITLE
Ratings for many episodes are missing.

### ISSUE THREAD

**Issue by BaukeZwart** [mynttt/UpdateTool#88]
All my shows (except one) have IMDB ratings now, but for quite a few shows the episode ratings are missing completely missing and for few series the episode raing is partly missing.
I have both the TVDB and TMDB api's in my my docker compose.

Ass an example for this serie I have ratings for the serie and episodes 1,2 and 3, the rest is missing.
[https://www.imdb.com/title/tt14354984/episodes?season=1&ref_=tt_eps_sn_1](url)

The shows with all episode rating missing appear to be mostly non english shows like this one.
[https://www.imdb.com/title/tt11009944/episodes?ref_=tt_eps_sm](url)

This this something that can be fixed?

**Comment by mynttt** [comment]
> This this something that can be fixed?

Depends. It would be good if you could send me your database via email (marc@herschel.io) so I can take a look at it. Then I can decide if it is an issue with UpdateTool or if Plex just supplies wonky metadata.

**Comment by BaukeZwart** [comment]
I think I found the issue, for example the first example with the ratings after e3 missing.
I checked TMDB and the IMDB ID for e4 and up was missing.
I added that, now I guess I need to wait 2 weeks to see if that fixed it.
If that is the issue I need to update TMDB for the missing IMDB ID's.

I can still send you the DB if you think that can be help full.

**Comment by mynttt** [comment]
Yeah, I've received your DB and looked at it. Like you describe, the first three items already have an IMDB associated within the Plex Database and are thus working fine.

The following items then only have TMDB/TVDB ids listed, which is where the issue begins. At the moment, the tool works in a way were always the TVDB id is associated with the entry. This is something that I will change in the future so both TVDB/TMDB ids are taken into consideration.

Within TVDB there are no ` "remoteIds": [], <- https://api4.thetvdb.com/v4/episodes/8901682/extended` entries for those episodes which leads to no resolved IMDB id and thus no updates for those items.

With TMDB there has always been an issue as they don't have a public facing endpoint for the strange IDs that Plex persists. Those IDs are in this format: `tmdb://3451276` and the public facing API requires the season number + episode nr in order with the root series id. The older agents supplied the IDs in this format: `tmdb://3451276/1/1` which has all the data that is required for it.

Luckily I've just found out that the series root item has a normal ID that can be resolved via TMDB => `tmdb://139492` and leads to a result when using their API: `https://api.themoviedb.org/3/tv/139492/season/1/episode/4/external_ids?api_key=xyz` (of course no IMDB added yet, which you would have to do with the TMDB website).

Case in point, the TMDB functionality of UpdateTool is currently unable to process items from new plex agent libraries. I will fix this however and internally reconstruct those legacy IDs with the episode and season data within the plex library so API calls for those items will be possible again.

Then TMDB functionality will also be supported for TV Show items of new plex library IDs and I can also add redundancy i.e. try TMDB first if no success try TVDB. This should allow for a better coverage of different metadata providers and also give you the chance to add those missing IMDB ids on TMDB so UpdateTool can pull them once TMDB publishes the changes.

**Comment by BaukeZwart** [comment]
Thanks for the detailed explanation.

**Comment by mynttt** [comment]
@BaukeZwart So I have an update on this situation. With today's update, at least the tool will now consider all available providers when resolving the external identifiers against an IMDB ID. For movies, it is with the priority TMDB / TVDB and for TV shows the same but reversed.

However, the issue with the new agent TMDB IDs for TV shows not being compatible with their API is still not resolved. I'll write them a mail later and ask them if they can extend their API, as that is the easiest way to deal with this. I ever did that with TVDB, and they were very compliant with my request. So right now even if there is only a TMDB ID or the TMDB ID has the correct IMDB linkage while TVDB does not have it, the tool is unable to resolve it as their public facing API can't deal with the IDs that Plex has persisted in its database.

**Comment by BaukeZwart** [comment]
Thanks for the update.
I will test the updated version.

**Comment by aussiehuddo** [comment]
@mynttt can you please tell me exactly where in the database the imdb id is stored? as in which table and column.

**Comment by MitjaHenner** [comment]
> @mynttt can you please tell me exactly where in the database the imdb id is stored? as in which table and column.
Did you find it? Looking to add the id to a few shows manually because they don't get updated for some reason.

**Comment by aussiehuddo** [comment]
> Did you find it? Looking to add the id to a few shows manually because they don't get updated for some reason.

Directly editing the database is not the way to achieve this, and in general a bad idea.

To ensure it gets updated add the IMDB ID's to the affected episodes/shows on theTVDB.com. Then the info will get pulled into Plex. This has the bonus of helping everyone with the same issue for those episodes.

**Comment by MitjaHenner** [comment]
> > Did you find it? Looking to add the id to a few shows manually because they don't get updated for some reason.
> 
> Directly editing the database is not the way to achieve this, and in general a bad idea.
> 
> To ensure it gets updated add the IMDB ID's to the affected episodes/shows on theTVDB.com. Then the info will get pulled into Plex. This has the bonus of helping everyone with the same issue for those episodes.

TVDB has IMDB linked, but the shows only show the TMDB rating. The episodes show the IMDB rating.
This only happens for two shows: Arcane and Queen's Gambit.
